### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by adaxi**
This pull request updates Nan to the latest version and fixes the deprecation warnings.

Note: I am a beginner at this. Tests pass. I am pretty confident in the patch, except at: 

https://github.com/justmoon/node-bignum/pull/105/files#diff-aea5ac9d873a4cf693091c97c0432e66R387

Perhaps someone more experience can tell me if what is did is enough.

### CODE REVIEW

**Comment by adaxi** [comment]
By the way, tests on 0.10 seem to fail because `standard` does not run on those versions.
Should we remove `standard` from the test list? Or ignore that 0.10 and 0.12 are failing?

**Comment by vladikoff** [comment]
@justmoon @rvagg is there time to look at this for node 8 and 10?

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 os:
   - linux
   - osx
+node_js:
+  - 0.10
+  - 0.12
+  - 4
+  - 6
```
Comment (vladikoff) [inline .travis.yml]: for next releases we should just bother with 6,8,10 IMHO and forget the old versions

Comment (adaxi) [reply .travis.yml]: I'll clean up this PR tomorrow, and remove unsupported LTS releases of node.

File: package.json [hunk 2]
```diff
@@ -22,18 +22,18 @@
     "url": "http://www.justmoon.net"
   },
   "dependencies": {
-    "nan": "^2.3.4",
+    "nan": "^2.9.2",
```
Comment (vladikoff) [inline package.json]: @adaxi does it work if you update to latest `nan` ?

**Comment by adaxi** [comment]
OpenSSL 1.1.0 is now bundled with Node 10 instead of OpenSSL 1.0.2, which breaks compatibility.

**Comment by vladikoff** [comment]
@rvagg pinging again. I'm guessing at some point this lib will be deprecated, once node 12 or whatever has BigInts?

**Comment by vladikoff** [comment]
So just need to maintain it a bit more :D

**Comment by vladikoff** [comment]
ping @rvagg

**Comment by rvagg** [comment]
I pulled in #112 for v10 support and to fix some of these problems. Have a go with that.

**Pull request closed without merging**